core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1054,6 +1054,11 @@ public RelBuilder scan(Iterable<String> tableNames) {
     final RelNode scan = scanFactory.createScan(cluster, relOptTable);
     push(scan);
     rename(relOptTable.getRowType().getFieldNames());
+
+    // When the node is not a TableScan but from expansion, we add the alias.
```

Comment:
I would rephrase a bit the comment line, for example something like:
"..., we need to explicitly add the alias."